src/sidebar/store/create-store.js

```diff
@@ -187,6 +187,22 @@ export function createStore(modules, initArgs = [], middleware = []) {
   return store;
 }
 
+/**
+ * Helper for creating an action which checks that the type of the action's
+ * payload is compatible with what the reducer expects.
+ *
+ * @template {ReducerMap<any>} Reducers
+ * @template {keyof Reducers} Type
+ * @param {Reducers} reducers - The map of reducer functions from a store module
+ * @param {Type} type - The name of a specific reducer in `reducers`
+ * @param {Parameters<Reducers[Type]>[1]} payload - The fields of the action
+ *   except for `type`. Pass `undefined` if the reducer doesn't need an action payload.
```

Comment:
OK, fresh brain today, I think I have bottomed out on where any remaining confusion lies here. I've understood what this line's typing means from a loose-human perspective through this whole endeavor: "The second parameter of the reducer function mapped to `type`", which is the payload.

I realize that where I'd been getting hung up was on the reference to `type` in the _comment_ for `payload`. It's _accurate_ from a redux-terminology perspective: the payload here is an `action` object minus the `type` property. But the term `type` is quite overloaded here:

* There's a `Type` generic and parameter already to _this_ function that is distinct from the `type` property of an `action` object and
* We don't include `type` in the actual action objects (payloads) passed to our reducers (or, at least I don't think we do?). Do we? 

In any case, I think this is/was the root of my confusion.